CreateSymLinkFolder.ps1

```diff
@@ -1,17 +1,203 @@
+using System.Reflection;
+using System.Windows.Forms;
+/// <summary>
+/// Present the Windows Vista-style open file dialog to select a folder. Fall back for older Windows Versions
+/// </summary>
+#pragma warning disable 0219, 0414, 0162
+public class FolderSelectDialog {
+    private string _initialDirectory;
+    private string _title;
+    private string _message;
+    private string _fileName = "";
+    
+    public string InitialDirectory {
+        get { return string.IsNullOrEmpty(_initialDirectory) ? Environment.CurrentDirectory : _initialDirectory; }
+        set { _initialDirectory = value; }
+    }
+    public string Title {
+        get { return _title ?? "Select a folder"; }
+        set { _title = value; }
+    }
+    public string Message {
+        get { return _message ?? _title ?? "Select a folder"; }
+        set { _message = value; }
+    }
+    public string FileName { get { return _fileName; } }
+
+    public FolderSelectDialog(string defaultPath="MyComputer", string title="Select a folder", string message=""){
+        InitialDirectory = defaultPath;
+        Title = title;
+        Message = message;
+    }
+    
+    public bool Show() { return Show(IntPtr.Zero); }
+
+    /// <param name="hWndOwner">Handle of the control or window to be the parent of the file dialog</param>
+    /// <returns>true if the user clicks OK</returns>
+    public bool Show(IntPtr? hWndOwnerNullable=null) {
+        IntPtr hWndOwner = IntPtr.Zero;
+        if(hWndOwnerNullable!=null)
+            hWndOwner = (IntPtr)hWndOwnerNullable;
+        if(Environment.OSVersion.Version.Major >= 6){
+            try{
+                var resulta = VistaDialog.Show(hWndOwner, InitialDirectory, Title, Message);
+                _fileName = resulta.FileName;
+                return resulta.Result;
+            }
+            catch(Exception){
+                var resultb = ShowXpDialog(hWndOwner, InitialDirectory, Title, Message);
+                _fileName = resultb.FileName;
+                return resultb.Result;
+            }
+        }
+        var result = ShowXpDialog(hWndOwner, InitialDirectory, Title, Message);
+        _fileName = result.FileName;
+        return result.Result;
+    }
+
+    private struct ShowDialogResult {
+        public bool Result { get; set; }
+        public string FileName { get; set; }
+    }
+
+    private static ShowDialogResult ShowXpDialog(IntPtr ownerHandle, string initialDirectory, string title, string message) {
+        var folderBrowserDialog = new FolderBrowserDialog {
+            Description = message,
+            SelectedPath = initialDirectory,
+            ShowNewFolderButton = true
+        };
+        var dialogResult = new ShowDialogResult();
+        if (folderBrowserDialog.ShowDialog(new WindowWrapper(ownerHandle)) == DialogResult.OK) {
+            dialogResult.Result = true;
+            dialogResult.FileName = folderBrowserDialog.SelectedPath;
+        }
+        return dialogResult;
+    }
```

Comment:
hm, ok, this is managable i suppose. don't quite get why, but not bad when all these unnecessary Wrappers are swallowed and we pretend as if they're not there.